container/docker/factory.go

```diff
@@ -197,6 +208,80 @@ func startThinPoolWatcher(dockerInfo *dockertypes.Info) (*devicemapper.ThinPoolW
 	return thinPoolWatcher, nil
 }
 
+func getKernelVersion() (string, error) {
+	unameOutput := syscall.Utsname{}
+	err := syscall.Uname(&unameOutput)
+	if err != nil {
+		return "", err
+	}
+
+	b := make([]byte, len(unameOutput.Release))
+	for i, v := range unameOutput.Release {
+		b[i] = byte(v)
+	}
+	return string(b), nil
+}
+
+func ensureThinLsKernelVersion(kernelVersion string) error {
+	// kernel 4.4.0 has the proper bug fixes to allow thin_ls to work without corrupting the thin pool
+	version_4_4_0 := semver.MustParse("4.4.0")
+	// RHEL kernel 3.10.0 release >= 366 has the proper bug fixes backported from 4.4.0 to allow
+	// thin_ls to work without corrupting the thin pool
+	version_3_10_0 := semver.MustParse("3.10.0")
+
+	matches := version_re.FindStringSubmatch(kernelVersion)
```

Comment:
I just tried Eric's suggestion, and because 4.4.0-abc has a pre-release component (abc), it is considered less than 4.4.0 proper. So that won't work.